txstratum/protocol.py

```diff
@@ -345,12 +345,26 @@ def is_mining_block(self) -> bool:
         return self.current_job.is_block
 
     async def refresh_job(self) -> None:
-        """Refresh miner's job, updating timestamp."""
+        """Refresh miner's job, updating timestamp.
+
+        This is run as a periodic task.
+        """
         assert self.current_job is not None
+
+        # If the miner is already mining a Tx, we do not interfere, otherwise we
+        # would be making it switch jobs unnecessarily.
+        if isinstance(self.current_job, MinerTxJob):
+            return
```

Comment:
What happens if the new job is also "mining a tx" but a different tx, what if the current_job was already found by another miner?

Reply:
> What happens if the new job is also "mining a tx" but a different tx

Yeah, we need to check this indeed. I'll change the implementation.

> what if the current_job was already found by another miner?

I think this would continue to work, since it is triggered in the manager method `stop_mining_tx`.
But we should discuss this after I commit the new implementation.

Reply:
Great!